packages/cosmos/src/authz.rs

```diff
@@ -42,54 +41,53 @@ pub struct MsgGrantHelper {
     pub expiration: Option<DateTime<Utc>>,
 }
 
-impl TryFrom<MsgGrantHelper> for TypedMessage {
-    type Error = anyhow::Error;
-
-    fn try_from(value: MsgGrantHelper) -> Result<Self> {
-        MsgGrant::try_from(value).map(TypedMessage::from)
+impl From<MsgGrantHelper> for TypedMessage {
+    fn from(value: MsgGrantHelper) -> Self {
+        MsgGrant::from(value).into()
     }
 }
 
-impl TryFrom<MsgGrantHelper> for MsgGrant {
-    type Error = anyhow::Error;
-
-    fn try_from(
+impl From<MsgGrantHelper> for MsgGrant {
+    fn from(
         MsgGrantHelper {
             granter,
             grantee,
             authorization,
             expiration,
         }: MsgGrantHelper,
-    ) -> Result<Self> {
+    ) -> Self {
         let authorization = GenericAuthorization { msg: authorization };
         let authorization = prost_types::Any {
             type_url: "/cosmos.authz.v1beta1.GenericAuthorization".to_owned(),
             value: authorization.encode_to_vec(),
         };
-        Ok(MsgGrant {
+        MsgGrant {
             granter: granter.get_address_string(),
             grantee: grantee.get_address_string(),
             grant: Some(Grant {
                 authorization: Some(authorization),
-                expiration: expiration.map(datetime_to_timestamp).transpose()?,
+                expiration: expiration.map(datetime_to_timestamp),
             }),
-        })
+        }
     }
 }
 
-fn datetime_to_timestamp(x: DateTime<Utc>) -> Result<Timestamp> {
-    Ok(prost_types::Timestamp {
+fn datetime_to_timestamp(x: DateTime<Utc>) -> Timestamp {
+    prost_types::Timestamp {
         seconds: x.timestamp(),
-        nanos: x.timestamp_subsec_nanos().try_into()?,
-    })
+        nanos: x
+            .timestamp_subsec_nanos()
+            .try_into()
+            .expect("DateTime<Utc>'s nanos is too large"),
```

Comment:
I guess you don't expect this to happen too often.

Reply:
More than that, it should be a straight up bug if this ever happens. It means that the `chrono` library is returning data outside valid bounds.